src/librustc_mir/transform/simplify_cfg.rs

```diff
@@ -118,12 +83,11 @@ impl<'tcx> MirPass<'tcx> for SimplifyCfg {
         while changed {
             pretty::dump_mir(tcx, "simplify_cfg", &counter, src, mir, None);
             counter += 1;
-            changed = self.simplify_branches(mir);
             changed |= self.remove_goto_chains(mir);
```

Comment:
Yes that's why I commented the pass out.
On May 19, 2016 3:39 PM, "Björn Steinbrink" <notifications@github.com>
wrote:

> In src/librustc_mir/transform/simplify_cfg.rs
> <https://github.com/rust-lang/rust/pull/33628#discussion_r63869926>:
>
> > @@ -118,12 +83,11 @@ impl<'tcx> MirPass<'tcx> for SimplifyCfg {
> >          while changed {
> >              pretty::dump_mir(tcx, "simplify_cfg", &counter, src, mir, None);
> >              counter += 1;
> > -            changed = self.simplify_branches(mir);
> >              changed |= self.remove_goto_chains(mir);
>
> This is an endless loop now, as changed is never set to false.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/33628/files/ee4a7c39e84237756fd453356eaec0921356873f#r63869926>
>